### PR TITLE
fix(adapter-ui): dispatch bound Actions from transition pills and kanban drops

### PR DESCRIPTION
**Pull request by laofahai**
## Summary
- Live-testing wave follow-up (flagged in #536): the **transition pills** (`transition-buttons.tsx`) and **kanban drag-to-column** (`auto-kanban.tsx`) surfaces called `transitionRecord` (raw status update) even when the state machine declared a bound Action — so declarative effects (`setFields` stamps, Flow triggers like purchase-approval) never fired from those surfaces. Core contract (`packages/core/src/types/state.ts`): transitions must be bound to Actions.
- New shared DI dispatcher `src/lib/transition-dispatch.ts`: `executeTransition` runs `executeAction(name, {id})` when the transition has a bound action (then re-queries the fresh record; falls back to full refetch when the re-query fails) and uses `transitionRecord` only for unbound transitions. `resolveBoundAction` maps a from/to edge to its declared action (string-or-array `from`, empty action treated as unbound).
- Both surfaces now dispatch through it; header buttons keep using the #536 `executeHeaderAction` path (comment-only touch-ups there).

## Evidence
Same bug class as #536, verified at file:line before fixing: pills discarded the non-optional `AvailableTransition.action`; kanban used `stateDefinition.transitions` only for validity and ignored the declared action.

## Testing
- 9 new dependency-injected tests (`transition-dispatch.test.ts`): bound→executeAction (never transitionRecord), re-query behavior, action failure surfaces server message, unbound fallback exact args, resolveBoundAction edge cases. No globalThis.fetch mocks.
- Full suite green (batched runner), typecheck + biome clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: addons/adapter-ui/cap-adapter-ui/src/lib/transition-dispatch.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+/**
+ * Resolve the Action bound to a state-machine transition edge (from → to).
+ * Returns undefined when no edge matches or the edge declares no action
+ * (empty action name) — callers then fall back to the raw transition mutation.
+ */
+export function resolveBoundAction(
+  transitions: readonly Transition[],
+  fromState: string,
+  toState: string,
+): string | undefined {
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/adapter-ui/cap-adapter-ui/src/lib/transition-dispatch.ts:55]: Fixed in 12950c0a — `resolveBoundAction` now accepts `undefined | null` transitions and returns `undefined` (raw-fallback path), with a regression test for both.

File: addons/adapter-ui/cap-adapter-ui/src/components/transition-buttons.tsx [hunk 2]
```diff
@@ -85,12 +95,28 @@ export function TransitionButtons({
...
+      });
+      if (outcome.kind === "failed") {
+        toast.error(outcome.message ?? t("toast.transitionFailed", "Status change failed"));
+        return;
+      }
       toast.success(t("toast.transitionSuccess", "Status changed successfully"));
-      onTransitioned?.(updated as Record<string, unknown>);
+      // updated: null → bound-action re-query failed; skip the in-place
+      // record refresh but still refresh the available transitions.
+      if (outcome.updated) onTransitioned?.(outcome.updated);
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/adapter-ui/cap-adapter-ui/src/components/transition-buttons.tsx]: Fixed in 12950c0a — `onTransitioned` is now called even when the re-query failed (`outcome.updated ?? undefined`), and its signature documents the undefined case so parents fall back to a full refetch. The kanban path already refetched unconditionally via `onTransitioned()`.
